laurent22/joplin
# Desktop: Resolves #9450: Make the beta markdown editor the default

# Summary

This pull request moves the CodeMirror 6-based markdown editor out of beta on desktop.

Related #9450, #10795.

# Handling incompatible plugins

Currently, the following plugins are marked as incompatible or mostly incompatible with CodeMirror 6, though there may be others:
- `com.septemberhx.Joplin.Enhancement`
   - Most features are incompatible with the CodeMirror 6-based editor.
- `ylc395.noteLinkSystem`
- `outline`
    - Jump to header may not work.
- `joplin.plugin.cmoptions`
- `com.ckant.joplin-plugin-better-code-blocks`
- `plugin.calebjohn.MathMode`
    - See https://github.com/CalebJohn/joplin-math-mode/pull/26

# Screenshots

<figure>
<img src="https://github.com/user-attachments/assets/35d8372b-8d62-4f5b-bff2-fa83c050517b" alt="" width="100%">
<figcaption>A notification, "The following plugins may not support the current markdown editor: list item Math Mode", offers a choice to "Switch to the legacy editor" or "Dismiss"</figcaption>
</figure>


---

<figure>
<img src="https://github.com/user-attachments/assets/6d883557-aecd-46f1-ad81-e5ca7a905d23" alt="" width="100%">
<figcaption>The "Enable spellcheck in the Markdown editor" setting is no longer in beta, "use the legacy Markdown editor" is included in the "general" tab.</figcaption>
</figure>

# Manual testing

1. Install the "math mode" plugin.
2. Install another plugin, if only one plugin is installed.
3. Open the editor.
4. Verify that the compatibility warning is shown.
5. Click "dismiss".
6. Verify that the compatibility warning is hidden.
7. Disable the plugin that is not "math mode" and restart Joplin.
8. Verify that the compatibility warning is still hidden.
9. Re-enable the plugin disabled in step 7 and restart Joplin.
10. Verify that the compatibility warning is still hidden.
11. Show the compatibility warning.
    - I did this from the development tools using `Setting.setValue('editor.pluginCompatibilityBannerDismissedFor', [])`
12. Click "Switch to the legacy editor".
13. Verify that the compatibility warning is no longer visible.
14. Verify that the legacy editor is enabled.
15. Disable the legacy editor in settings.
16. Verify that the compatibility warning is visible.

This has been tested successfully on Ubuntu 24.04.
